File: redo/osxaltest/label.m
// 31 july 2015
#import "osxaltest.h"

@implementation tLabel {
	NSTextField *t;
	id<tControl> parent;
	NSLayoutPriority horzpri, vertpri;
}

- (id)init
{
	self = [super init];
	if (self) {
		self->t = [[NSTextField alloc] initWithFrame:NSZeroRect];
		[self->t setStringValue:@"Label"];
		[self->t setEditable:NO];
		[self->t setSelectable:NO];
		[self->t setDrawsBackground:NO];
		[self->t setFont:[NSFont systemFontOfSize:[NSFont systemFontSizeForControlSize:NSRegularControlSize]]];
		[self->t setBordered:NO];
		[self->t setBezelStyle:NSTextFieldSquareBezel];
		[self->t setBezeled:NO];
		[[self->t cell] setLineBreakMode:NSLineBreakByClipping];
		[[self->t cell] setScrollable:YES];
		[self->t setTranslatesAutoresizingMaskIntoConstraints:NO];

		self->parent = nil;

		self->horzpri = [self->t contentHuggingPriorityForOrientation:NSLayoutConstraintOrientationHorizontal];
		self->vertpri = [self->t contentHuggingPriorityForOrientation:NSLayoutConstraintOrientationVertical];
	}
	return self;
}

- (void)tSetParent:(id<tControl>)p addToView:(NSView *)v relayout:(BOOL)relayout
{
	self->parent = p;
	[v addSubview:self->t];
	if (relayout)
		[self tRelayout];
}

- (void)tFillAutoLayout:(tAutoLayoutParams *)p
{
	// reset the hugging priority
	[self->t setContentHuggingPriority:self->horzpri forOrientation:NSLayoutConstraintOrientationHorizontal];
	[self->t setContentHuggingPriority:self->vertpri forOrientation:NSLayoutConstraintOrientationVertical];
	p->view = self->t;
	p->attachLeft = YES;
	p->attachTop = YES;
	p->attachRight = YES;
	p->attachBottom = YES;
}

- (void)tRelayout
{
	if (self->parent != nil)
		[self->parent tRelayout];
}

@end

File: redo/osxaltest/label.swift
// 31 july 2015
import Cocoa

class Label : NSTextField, Control {
	private var parent: Control?

	init() {
		var cell: NSTextFieldCell

		self.parent = nil

		super.init(frame: NSZeroRect)
		self.stringValue = "Label"
		self.editable = false
		self.selectable = false
		self.drawsBackground = false
		self.font = NSFont.systemFontOfSize(NSFont.systemFontSizeForControlSize(NSControlSize.RegularControlSize))
		self.bordered = false
		self.bezelStyle = NSTextFieldBezelStyle.SquareBezel
		self.bezeled = false
		cell = self.cell() as! NSTextFieldCell
		cell.lineBreakMode = NSLineBreakMode.ByClipping
		cell.scrollable = true
		self.translatesAutoresizingMaskIntoConstraints = false
	}

	required init?(coder: NSCoder) {
		fatalError("can't use this constructor, sorry")
	}

	func View() -> NSView {
		return self
	}

	func SetParent(p: Control) {
		self.parent = p
	}
}
